src/Microsoft.CodeQuality.Analyzers/Core/Maintainability/ReviewUnusedParameters.Fixer.cs

```diff
@@ -22,9 +30,144 @@ public sealed override FixAllProvider GetFixAllProvider()
+
+            // Achieved the root and still could not find the node with parameters.
+            // Need to cancel the action.
+            return null;
+        }
+
+        private async Task<ImmutableArray<KeyValuePair<DocumentId, SyntaxNode>>> GetNodesToRemoveAsync(Document document, Diagnostic diagnostic, CancellationToken cancellationToken)
+        {
+            SyntaxNode root = await document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+            SyntaxNode node = root.FindNode(diagnostic.Location.SourceSpan);
+            node = GetParameterNode(node);
+
+            DocumentEditor editor = await DocumentEditor.CreateAsync(document, cancellationToken).ConfigureAwait(false);
+            ISymbol parameterSymbol = editor.SemanticModel.GetDeclaredSymbol(node);
+            ISymbol methodDeclarationSymbol = parameterSymbol.ContainingSymbol;
+
+            if (!IsSafeMethodToRemoveParameter(methodDeclarationSymbol))
+            {
+                // See https://github.com/dotnet/roslyn-analyzers/issues/1466
+                return ImmutableArray<KeyValuePair<DocumentId, SyntaxNode>>.Empty;
+            }
 
+            var nodesToRemove = ImmutableArray.CreateBuilder<KeyValuePair<DocumentId, SyntaxNode>>();
+            nodesToRemove.Add(new KeyValuePair<DocumentId, SyntaxNode>(document.Id, node));
+            var referencedSymbols = await SymbolFinder.FindReferencesAsync(methodDeclarationSymbol, document.Project.Solution, cancellationToken).ConfigureAwait(false);
+
+            foreach (var referencedSymbol in referencedSymbols)
+            {
+                if (referencedSymbol.Locations != null)
+                {
+                    foreach (var referenceLocation in referencedSymbol.Locations)
+                    {
+                        Location location = referenceLocation.Location;
+                        var referenceRoot = location.SourceTree.GetRoot();
+                        var referencedSymbolNode = referenceRoot.FindNode(location.SourceSpan);
+                        DocumentEditor localEditor = await DocumentEditor.CreateAsync(referenceLocation.Document, cancellationToken).ConfigureAwait(false);
+                        var operation = localEditor.SemanticModel.GetOperation(referencedSymbolNode, cancellationToken);
+                        var arguments = GetOperationArguments(referencedSymbolNode, localEditor.SemanticModel, cancellationToken);
+
+                        if (arguments != null)
+                        {
+                            foreach (IArgumentOperation argument in arguments)
+                            {
+                                if (argument.Parameter.Equals(parameterSymbol) && argument.ArgumentKind == ArgumentKind.Explicit)
+                                {
+                                    nodesToRemove.Add(new KeyValuePair<DocumentId, SyntaxNode>(referenceLocation.Document.Id, referenceRoot.FindNode(argument.Syntax.GetLocation().SourceSpan)));
```

Comment:
@ivanbasov Can we execute these fixer on big projects such as Roslyn and verify the experience, silent breaks, introduced errors due to the fact that GetOperationArguments returns null for some references, etc.? I think we should do some more testing before we add this fixer to our NuGet package, so probably this should go into our 2.7.0 package, not the 2.6.0 planned for this week. @jinujoseph thoughts?